core/src/main/java/com/datastax/oss/driver/internal/core/metadata/SchemaAgreementChecker.java

```diff
@@ -0,0 +1,200 @@
+import com.datastax.oss.driver.api.core.metadata.NodeState;
+import com.datastax.oss.driver.internal.core.adminrequest.AdminRequestHandler;
+import com.datastax.oss.driver.internal.core.adminrequest.AdminResult;
+import com.datastax.oss.driver.internal.core.adminrequest.AdminRow;
+import com.datastax.oss.driver.internal.core.channel.DriverChannel;
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.datastax.oss.driver.internal.core.util.NanoTime;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class SchemaAgreementChecker {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SchemaAgreementChecker.class);
+  private static final int INFINITE_PAGE_SIZE = -1;
+  @VisibleForTesting static final InetAddress BIND_ALL_ADDRESS;
+
+  static {
+    try {
+      BIND_ALL_ADDRESS = InetAddress.getByAddress(new byte[4]);
+    } catch (UnknownHostException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private final DriverChannel channel;
+  private final InternalDriverContext context;
+  private final int port;
+  private final String logPrefix;
+  private final Duration queryTimeout;
+  private final long intervalNs;
+  private final long timeoutNs;
+  private final long start;
+  private final CompletableFuture<Boolean> result = new CompletableFuture<>();
+
+  SchemaAgreementChecker(
+      DriverChannel channel, InternalDriverContext context, int port, String logPrefix) {
+    this.channel = channel;
+    this.context = context;
+    this.port = port;
+    this.logPrefix = logPrefix;
+    DriverConfigProfile config = context.config().getDefaultProfile();
+    this.queryTimeout = config.getDuration(CoreDriverOption.CONTROL_CONNECTION_TIMEOUT);
+    this.intervalNs =
+        config.getDuration(CoreDriverOption.CONTROL_CONNECTION_AGREEMENT_INTERVAL).toNanos();
+    this.timeoutNs =
+        config.getDuration(CoreDriverOption.CONTROL_CONNECTION_AGREEMENT_TIMEOUT).toNanos();
+    this.start = System.nanoTime();
+  }
+
+  public CompletionStage<Boolean> run() {
+    LOG.debug("[{}] Checking schema agreement", logPrefix);
+    if (timeoutNs == 0) {
+      result.complete(false);
+    } else {
+      sendQueries();
+    }
+    return result;
+  }
+
+  private void sendQueries() {
+    long elapsedNs = System.nanoTime() - start;
+    if (elapsedNs > timeoutNs) {
+      LOG.warn("[{}] Schema agreement not reached after {}", logPrefix, NanoTime.format(elapsedNs));
+      result.complete(false);
```

Comment:
Noticed while testing that the checker keeps trying even after the future completes,  shouldn't the code also return here?  Otherwise it keeps trying to verify whether or not the schema agrees.

I.e., default timeout set to 10s, but i noticed logs well after:

```
14:45:27.949 [c0-io-0] WARN  c.d.o.d.i.c.m.SchemaAgreementChecker - [c0] Schema agreement not reached after 22 s
```

Reply:
Right that's a mistake, the rest of them method should be in an `else` block.